src/app/pages/sharing/nvme-of/nvme-of.store.ts

```diff
@@ -0,0 +1,86 @@
+import { computed, Injectable } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import {
+  combineLatest, switchMap, tap,
+} from 'rxjs';
+import {
+  NvmeOfNamespace, NvmeOfSubsystem, SubsystemHostAssociation, SubsystemPortAssociation,
+} from 'app/interfaces/nvme-of.interface';
+import { ApiService } from 'app/modules/websocket/api.service';
+
+export interface NvmeOfState {
+  subsystems: NvmeOfSubsystem[];
+  namespaces: NvmeOfNamespace[];
+  hostSubsystems: SubsystemHostAssociation[];
+  ports: SubsystemPortAssociation[];
+  isLoading: boolean;
+}
+
+const initialState: NvmeOfState = {
+  subsystems: [],
+  namespaces: [],
+  hostSubsystems: [],
+  ports: [],
+  isLoading: false,
+};
+
+@Injectable()
+export class NvmeOfStore extends ComponentStore<NvmeOfState> {
+  readonly subsystems = computed(() => {
+    const subsystems = [...this.state().subsystems];
```

Comment:
Let's not copy objects if we don't have to. This may hide a problem with how we work with objects downstream.